instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/OutputStreamSnippetInjectionHelper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+import static java.util.logging.Level.FINE;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.logging.Logger;
+
+public class OutputStreamSnippetInjectionHelper {
+
+  private static final Logger logger =
+      Logger.getLogger(OutputStreamSnippetInjectionHelper.class.getName());
+
+  private final String snippet;
+
+  public OutputStreamSnippetInjectionHelper(String snippet) {
+    this.snippet = snippet;
+  }
+
+  /**
+   * return true means this method performed the injection, return false means it didn't inject
+   * anything Servlet3OutputStreamWriteAdvice would skip the write method when the return value is
+   * true, and would write the original bytes when the return value is false.
+   */
+  public boolean handleWrite(
+      InjectionState state, OutputStream out, byte[] original, int off, int length)
+      throws IOException {
+    if (state.isHeadTagWritten()) {
+      return false;
+    }
+    int i;
```

Comment:
Nit: can you rename this to `endOfHeadTagPosition` or sth similar? Since it's used later in this method